numpy/_core/src/multiarray/ctors.c

```diff
@@ -1541,12 +1543,19 @@ PyArray_FromAny(PyObject *op, PyArray_Descr *newtype, int min_depth,
 /*
  * Internal version of PyArray_FromAny that accepts a dtypemeta. Borrows
  * references to the descriptor and dtype.
+ *
+ * The `was_scalar` output returns 1 when the object was a "scalar".
+ * This means it was:
+ * - Recognized as a scalar by a/the dtype.  This can be DType specific,
+ *   for example a tuple may be a scalar, but only for structured dtypes.
+ * - Anything not recognized a scalar (mapped to a DType) but also not
```

Comment:
```suggestion
 * - Anything not recognized as an instance of a DType's scalar type but also not
```

I got tripped up on the original wording - take it or leave it, but if you leave it, it should be `recognized as a scalar`